### PR TITLE
sorbet/tapioca/require: make aware of platform-specific gem resolution

### PR DESCRIPTION
**Pull request by Bo98**
Previously when running Tapioca, we would have it `require` every gem in the lockfile.

This is somewhat correct except that it did not consider that some gems might not be installed on some platforms. Notably, we now have prebuilt gems for Nokogiri working with the Bundler 2 move so the `mini_portile2` gem is no longer installed except on platforms where there are no prebuilt gems. This would lead to Tapioca erroring when it can't find `mini_portile2`.

I've fixed the logic to instead consider only the set of gem specifications that apply to the local platform. This technically leads to a discrepancy between platforms but in practice, based on how our lockfile looks at the moment, the only difference currently is 32-bit arm Linux, where `brew typecheck (--update)` isn't supported anyway, plus those differences would have been even less minor than how Bundler's standalone `setup.rb` generation differs between every platform and would not have changed any typechecking result in the end.

TL;DR: fixes `brew typecheck --update` not working after Bundler 2 update

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Great work/find!

**Comment by Bo98** [comment]
btw, update on Dependabot: I sent a PR to them fixing the issue we had were it hasn't been working and it's merged now so we should hopefully be getting Dependabot PRs again very soon.
